api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/fields/SelectListGeneratorVTwo.java

```diff
@@ -54,103 +55,103 @@
-				return -1;
-			} else {
-				if ("".equals(s1[0])) {
-					return -1;
-				} else if ("".equals(s2[0])) {
-					return 1;
-				}
-				if (s2[1]==null) {
-					return 1;
-				} else if (s1[1] == null){
-					return -1;
-				} else {
-					return collator.compare(s1[1],s2[1]);
-				}
-			}
-		}
-	}
+    public static Map<String,String> getOptions(
+            EditConfigurationVTwo editConfig,
+            String fieldName,
+            VitroRequest vreq){
+
+
+        if( editConfig == null ){
+            log.error( "fieldToSelectItemList() must be called with a non-null EditConfigurationVTwo ");
+            return Collections.emptyMap();
+        }
+        if( fieldName == null ){
+            log.error( "fieldToSelectItemList() must be called with a non-null fieldName");
+            return Collections.emptyMap();
+        }
+
+        FieldVTwo field = editConfig.getField(fieldName);
+        if (field==null) {
+            log.error("no field \""+fieldName+"\" found from editConfig.");
+            return Collections.emptyMap();
+        }
+
+        if( field.getFieldOptions() == null ){
+            return Collections.emptyMap();
+        }
+
+        try {
+            //UQAM need vreq instead of WebappDaoFactory
+            Map<String, String> parentClass = Collections.emptyMap();
+            FieldOptions fieldOptions = field.getFieldOptions();
+            // UQAM TODO - Only internationalization of ChildVClassesWithParent are implemented. For TODO, implement the internationalization for the rest of instanceof "FieldOptions"
+            if (fieldOptions instanceof ChildVClassesWithParent ) {
+                return ((ChildVClassesWithParent)fieldOptions).getOptions(editConfig,fieldName,vreq);
+            } else {
+                return fieldOptions.getOptions(editConfig,fieldName,vreq.getWebappDaoFactory());
+            }
+        } catch (Exception e) {
+            log.error("Error runing getFieldOptionis()",e);
+            return Collections.emptyMap();
+        }
+    }
+
+
+    //Methods to sort the options map
+    // from http://forum.java.sun.com/thread.jspa?threadID=639077&messageID=4250708
+    //Modified to allow for a custom comparator to be sent in, defaults to mapPairsComparator
+    public static Map<String,String> getSortedMap(Map<String,String> hmap,
+            Comparator<String[]> comparator, VitroRequest vreq){
+        // first make temporary list of String arrays holding both the key and its corresponding value, so that the list can be sorted with a decent comparator
+        List<String[]> objectsToSort = new ArrayList<String[]>(hmap.size());
+        for (String key:hmap.keySet()) {
+            String[] x = new String[2];
+            x[0] = key;
+            x[1] = hmap.get(key);
+            objectsToSort.add(x);
+        }
+
+        //if no comparator is passed in, utilize MapPairsComparator
+        if(comparator == null) {
+            comparator = new MapPairsComparator(vreq);
```

Comment:
I changed the target of this pull-request from `master` to `sprint-i18n-whitespace`. Nothing additional needs to be done for that.

I then locally pulled down this pull-request to check the corrections with the following command:
```
git diff pull/150 master api/src/main/java/edu/cornell/mannlib/vitro/webapp/edit/n3editing/VTwo/fields/SelectListGeneratorVTwo.java
```
The result of that command shows that your updates are perfect... except in line:116 (here), and lines 142-146 below.

![Screenshot from 2020-04-02 10-30-59](https://user-images.githubusercontent.com/149283/78261386-2c629f80-74cd-11ea-8245-7a3337bd0bcd.png)